client/js/shout.js

```diff
@@ -183,14 +183,42 @@ $(function() {
 	});
 
 	socket.on("msg", function(data) {
+		var type = data.msg.type;
 		var target = "#chan-" + data.chan;
-		if (data.msg.type === "error") {
+		if (type === "error") {
 			target = "#chan-" + chat.find(".active").data("id");
 		}
 
 		var chan = chat.find(target);
 		var from = data.msg.from;
 
+		if ([
+			"join",
+			"mode",
+			"kick",
+			"nick",
+			"part",
+			"quit",
+			"topic",
+			"action",
+		].indexOf(type) !== -1) {
+			switch (type) {
+			case "join": data.msg.formattedAction = "has joined the channel"; break;
+			case "mode": data.msg.formattedAction = "sets mode"; break;
+			case "kick": data.msg.formattedAction = "has kicked"; break;
+			case "nick": data.msg.formattedAction = "is now known as"; break;
+			case "part": data.msg.formattedAction = "has left the channel"; break;
+			case "quit": data.msg.formattedAction = "has quit"; break;
+			case "topic": data.msg.formattedAction = "has changed the topic to:"; break;
```

Comment:
From my experience, topics are often long, full of special characters (colons, dashes, ...), like:

<img width="813" alt="screen shot 2016-01-19 at 23 59 06" src="https://cloud.githubusercontent.com/assets/113730/12440492/fbda6e24-bf09-11e5-85bd-bbdb985a0815.png">

What about displaying the new topic on its own line, like the following:

<img width="817" alt="screen shot 2016-01-20 at 00 03 07" src="https://cloud.githubusercontent.com/assets/113730/12440493/fbdb1d4c-bf09-11e5-96a3-2dc235759b4c.png">

Actually, changing topic is often a big deal that needs to be seen from everyone: I would display it differently to catch the eye, like in bold, or italic, ... What do you think?